### PR TITLE
hatch-vcs-tunable: feat: new project

### PR DESCRIPTION
**Pull request by sfoster1**
The hatch-vcs-tunable hook allows you to override settings passed to hatch_vcs (and to setuptools_scm from there) via environment variables, useful when working on a project that has multiple release tracks with different versions.

### CODE REVIEW

File: hatch-vcs-tunable/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# hatch-vvcs-tunable
```
Comment (SyntaxColoring) [inline hatch-vcs-tunable/README.md]:
```suggestion
# hatch-vcs-tunable
```

File: hatch-vcs-tunable/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+[![PyPI - Version](https://img.shields.io/pypi/v/hatch-git-version-tunable.svg)](https://pypi.org/project/hatch-git-version-tunable)
+[![PyPI - Python Version](https://img.shields.io/pypi/pyversions/hatch-git-version-tunable.svg)](https://pypi.org/project/hatch-git-version-tunable)
+[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)
+[![Checked with mypy](http://www.mypy-lang.org/static/mypy_badge.svg)](http://mypy-lang.org/)
+
+
+-----
+
+This is a plugin for [Hatch](https://github.com/pypa/hatch) that extends the plugin [hatch-vcs](https://github.com/ofek/hatch-vcs) to allow for overriding some config elements with environment variables.
```
Comment (SyntaxColoring) [inline hatch-vcs-tunable/README.md:11]: In the long run, should we aim to upstream this work into `hatch-vcs`?

File: hatch-vcs-tunable/src/hatch_vcs_tunable/version_source.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# SPDX-License-Identifier: Apache-2.0
+"""A version source extending hatch_vcs with the ability to set configuration from the environment."""
+from os import environ
+from collections import deque
+from typing import Iterable
+
+from .const import (
+    TAG_PATTERN_ENVIRON_NAME,
+    FALLBACK_VERSION_ENVIRON_NAME,
+    PLUGIN_NAME as _PLUGIN_NAME,
```
Comment (SyntaxColoring) [inline hatch-vcs-tunable/src/hatch_vcs_tunable/version_source.py:12]: What's this one rename `PLUGIN_NAME as _PLUGIN_NAME` for?

File: hatch-vcs-tunable/tests/__pycache__/__init__.cpython-310.pyc
Comment (SyntaxColoring) [inline hatch-vcs-tunable/tests/__pycache__/__init__.cpython-310.pyc:1]: Looks like these .pyc files slipped past our .gitignore somehow.

**Review by SyntaxColoring — APPROVED** [review]
Cool!

File: hatch-vcs-tunable/src/hatch_vcs_tunable/version_source.py (hunk 3 above)
Comment (sfoster1) [reply hatch-vcs-tunable/src/hatch_vcs_tunable/version_source.py:12]: so it's not shadowed by the mandatory name of the classmember

File: hatch-vcs-tunable/tests/test_version_source.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    subject: TunableVCSVersionSource, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    monkeypatch.setenv(
+        "HATCH_VCS_TUNABLE_RAW_OPTIONS", "my-key-1=my;val;1;my-key-2=my;val;2"
+    )
+    assert subject.config_raw_options == {
+        "upstream-key-1": "upstream-val-1",
+        "upstream-key-2": "upstream-val-2",
+        "my-key-1": "my;val;1",
+        "my-key-2": "my;val;2",
```
Comment (sanni-t) [inline hatch-vcs-tunable/tests/test_version_source.py:90]: Interesting!

**Review by sanni-t — APPROVED** [review]
Yay!

File: hatch-vcs-tunable/tests/test_version_source.py (hunk 4 above)
Comment (sfoster1) [reply hatch-vcs-tunable/tests/test_version_source.py:90]: This is really more a "seems like it should handle this" thing, I don't think in practice these options ever contain this data

**Pull request merged**